Added cloneopts to ice modifiers to pass git clone flags.

jamartin9 opened the pull request:
Added cloneopts to ice modifiers to pass git clone flags.
Changed clone depth to 1.
ex.
```
zplugin ice cloneopts'--branch develop --single-branch'
zplugin light zsh-users/zsh-autosuggestions
```
Edit: If this is unwanted or requires a better implementation feel free to close the PR.

psprint commented:
It's an interesting solution to the use case of fetching a specific version of plugin. I think it's better than the abstract one, i.e. doing this via ice `branch"develop"`, which requires user to do in-mind mapping onto actual git command, and isn't clear if other branches will get downloaded, etc.

I will merge, but could you leave the `--depth 10` in `doc/install.sh` and `doc/mod-install.sh` ? Also, `cloneopts''` ice should be added to 4 places, after `teleid''`:

```zsh
# zplugin-autoload.zsh, line 1426

    # Read disk-Ice
    local -A __mdata
    local __key
    { for __key in mode url from as pick bpick mv cp make atclone atpull is_release ver id-as teleid; do
        [[ -f "$__zplugin_path/$__key" ]] && __mdata[$__key]="$(<$__zplugin_path/$__key)"
    ....

# zplugin-autoload.zsh, line 1447

    # Final decision, static ice vs. saved ice
    local -A __MY_ICE
    for __key in mode url from as pick bpick mv cp atclone atpull is_release ver id-as teleid   make svn; do
        (( ${+__sice[$__key]} + ${${${__pack:#pack-nf}:+${+__mdata[$__key]}}:-0} )) && __MY_ICE[$__key]="${__sice[$__key]-${__>
    done

# zplugin-autoload.zsh, line 2511

-zplg-recall() {
    local -A ice
    local el val cand1 cand2 local_dir filename

    local -a ice_order nval_ices output
    ice_order=( svn proto from teleid id-as depth if wait load unload blockf pick bpick src as ver silent mv cp atinit atclone>
    nval_ices=( blockf silent svn )
    ....

# zplugin-side.zsh, line 169

-zplg-store-ices() {
    local __pfx="$1" __ice_var="$2" __add_ices="$3" __add_ices2="$4" url="$5" mode="$6"

    command mkdir -p "$__pfx"
    local __key __var_name
    for __key in proto from as bpick mv cp atclone atpull ver id-as teleid ${(s: :)__add_ices[@]}; do
        __var_name="${__ice_var}[$__key]"
        print -r -- "${(P)__var_name}" >! "$__pfx"/$__key

```

jamartin9 commented:
Thank you for the the feedback. I reverted the `--depth=1` change in the install scripts. 
I have added `cloneopts` to the places you mentioned after `teleid`. I apologize for missing those areas as I was essentially trying to copy `depth`  (which doesn't appear in several). 

I prefer allowing a low abstraction interface for power users with abstracted convenience layers for generalized easy use. This change does raise a concern for me with the`depth` flag having potential overlap with `cloneopts`, i.e. specifying `--depth` twice. I assume users are knowledgeable enough to not overload it or there could be additional checks/consolidation added in the future if the issue becomes prevalent. 

Any additional thoughts/feedback are always welcome too.

psprint commented:
The tests are failing, it's a standard problem that a new stored ice'' modifier adds a file to each test result, and `diff -r model-dir answer-dir` reports difference (return status 1), and test is reported to fail.

Let this be a lession that following should be ran before merging/submitting:

```zsh
cd zplugin.git
git worktree add test tests # tests are in separate orphan branch "tests", checkout them as an additional work-dir "test"
make alltest
```

I'll fix the tests later today.
